### PR TITLE
[II] Load Brandon GLM-5.2 SQG W4A8

### PR DESCRIPTION
**Pull request by voipmonitor**
## Status

**Implemented and qualified with the B12X SQG runtime.** Human review is
required before merge.

## Purpose

Load and serve `brandonmusic/GLM-5.2-SQG-W4A8` revision
`593dd0d2de6f79ce4e65303930c22c75e1359d44`. The checkpoint uses the SQG
reconstruction primitive from the QSRT/KQuant stack, with Brandon-specific
BMM-Law calibration and independent K3/K4/K6 allocation. It is not a canonical
paired-rate QSRT checkpoint.

## Resulting behavior

- Validate the explicit `glm52_sqg_atoms_v2_w4a8_v1` metadata contract before
  model construction.
- Stream-slice topology-neutral routed payloads for tensor parallel loading,
  preserving independent gate/up/down K3/K4 assignments.
- Dispatch routed experts to B12X W4A8 and uniform non-routed K6 matrices to
  B12X W6A16. Compatible K6 gate/up projections share one prepared launch.
- Plan one or two graph-owned runtime arenas according to the scheduler's
  logical microbatch count; capture streams do not duplicate model scratch.
- Reject A16 fallback, incomplete layer inventories, mixed codebook markers,
  unsupported TP/DCP geometry, and unplanned CUDA graph capacity.
- Optionally write per-rank load and execution closure to
  `VLLM_GLM_SQG_W4A8_EVIDENCE_DIR` for release qualification.
- Preserve the projection-mixed R7, MCG, online EXL3, and canonical QSRT
  execution contracts.

## Dependency and scope

This PR is based on vLLM PR #300 because the checkpoint reuses its
projection-mixed tensor loader and stream-time TP slicing. The diff against
`agent/pr300-base-glm52-sqg`, a read-only mirror of PR #300 head
`c6b4fa5ec0dfa7fb62d1ba81da11071b95948be2`, contains only the SQG checkpoint
contract, native dispatch, graph ownership, evidence output, and tests. Merge
#300 first, then retarget this PR to `dev/infernal-invocation` and delete the
mirror branch.

This does not duplicate PR #269, which loads canonical Fruit QSRT atoms and
targets a different checkpoint schema and execution contract.

## Validation

- vLLM EXL3 and environment suites: **144 passed** in 70.89 s.
- B12X CPU/source contract suite: **51 passed, 34 GPU tests skipped**.
- B12X SM120 suite: **85 passed** in 117.29 s, including exhaustive K6 decode,
  numerical references, fixed scratch bounds, and CUDA graph replay.
- TP4/DCP1/MTP0 full-graph serving on root-port RTX PRO 6000 Blackwell GPUs:
  **31.3 tok/s C1 decode** and **2,856 tok/s 8K prefill**.
- Teacher-forced KLD against official BF16 logits: **0.08101 with FP8 KV** and
  **0.12375 with NVFP4 KV**; BF16-equivalent KV diagnostic: **0.07865**.

AI assistance: OpenAI Codex was used for implementation, profiling, tests, and
PR drafting. A human maintainer must review and defend the change before merge.

### CODE REVIEW

**Comment by voipmonitor** [comment]
## Qualified release evidence

**Status: qualified** for `brandonmusic/GLM-5.2-SQG-W4A8` revision `593dd0d2de6f79ce4e65303930c22c75e1359d44` in TP4/DCP1/MTP0 with native B12X SQG execution and NVFP4 DS-MLA KV.

The immutable release image contains this pull request's head and no private source patch:

```text
voipmonitor/vllm:infernal-invocation-vllm0af7310-b12xec2f97d-fi1ac6942-cu133-torch213-20260814-r13
sha256:a9059b81458167ffd27df77c17e928af81650404eff3e7f90ad528a16c96b10b
```

Validation:

- 144 focused vLLM loader and environment tests passed.
- All four rank-evidence records listed routed layers 3 through 77 as loaded and executed through `full-w4a8`; A16 fallback was disabled.
- FULL and PIECEWISE CUDA graphs captured rows 1 through 6.
- Two warmed 512-token CC1 runs reached 31.02 and 31.26 tok/s.
- Three uncached 8,192-token prefills reached 3,040.9, 3,025.3, and 3,043.7 tok/s.
- The server remained healthy and produced the expected arithmetic result.

Runbook: https://github.com/local-inference-lab/rtx6kpro/blob/master/models/glm5.2-infernal-invocation-r13-sqg.md

Merge contract: https://github.com/local-inference-lab/rtx6kpro/issues/67

This pull request remains draft because its comparison base is the exact head of vLLM PR #300. Merge #300 into `dev/infernal-invocation`, retarget this pull request to that branch, and verify that the remaining semantic diff is limited to GLM SQG schema and runtime support before merge.

**Comment by voipmonitor** [comment]
The `brandonmusic/GLM-5.2-SQG-W4A8` checkpoint is no longer part of the Infernal Invocation DS4F and GLM release contract. Its required B12X implementation in local-inference-lab/b12x#197 is closed and incompatible with the B12X master execution contract. No canonical vLLM behavior depends on this draft. Historical Infernal Invocation r13 source locks and qualification artifacts remain unchanged as reproducible records.
